node/worker/worker.go

```diff
@@ -130,23 +130,44 @@ func (w *Worker) CommitTransactions(
 	// STAKING - only beaconchain process staking transaction
 	if w.chain.ShardID() == shard.BeaconChainShardID {
 		for _, tx := range pendingStaking {
+			// TODO: merge staking transaction processing with normal transaction processing.
+			// <<THESE CODE ARE DUPLICATED AS ABOVE
+			// If we don't have enough gas for any further transactions then we're done
+			if w.current.gasPool.Gas() < params.TxGas {
+				utils.Logger().Info().Uint64("have", w.current.gasPool.Gas()).Uint64("want", params.TxGas).Msg("Not enough gas for further transactions")
+				break
+			}
+			// Check whether the tx is replay protected. If we're not in the EIP155 hf
+			// phase, start ignoring the sender until we do.
+			if tx.Protected() && !w.config.IsEIP155(w.current.header.Number()) {
+				utils.Logger().Info().Str("hash", tx.Hash().Hex()).Str("eip155Epoch", w.config.EIP155Epoch.String()).Msg("Ignoring reply protected transaction")
+				txs.Pop()
+				continue
+			}
+
+			// Start executing the transaction
+			w.current.state.Prepare(tx.Hash(), common.Hash{}, len(w.current.txs))
```

Comment:
there is another Prepare() call above right? why is this needed?

Reply:
It is needed for every transactions to setup the context.